nixos/tests/nextflow.nix

```
@@ -0,0 +1,60 @@
+import ./make-test-python.nix (
+  { pkgs, ... }:
+  let
+    bash = pkgs.dockerTools.pullImage {
+      imageName = "quay.io/nextflow/bash";
+      imageDigest = "sha256:bea0e244b7c5367b2b0de687e7d28f692013aa18970941c7dd184450125163ac";
+      sha256 = "161s9f24njjx87qrwq0c9nmnwvyc6iblcxka7hirw78lm7i9x4w5";
```

This hash is different for aarch64-linux apparently (quoting from ofborg): 

```
error: hash mismatch in fixed-output derivation '/nix/store/75i49g2lqizqy7wynln75imsw7dhznfx-docker-image-quay.io-nextflow-bash-latest.tar.drv':
         specified: sha256-hZOe4qkUHZ4jPGp2Rlc0zG9uq00MYJ7xQV1KS4RLOpg=
            got:    sha256-oUiM5XRHWbogXfjmSMDaKvG2ZQy0d4GziFWyLFWc0C4=
```

I wonder if we should simply do this:

```suggestion
      sha256 = {
        x86_64-linux = "161s9f24njjx87qrwq0c9nmnwvyc6iblcxka7hirw78lm7i9x4w5";
        aarch64-linux = "sha256-oUiM5XRHWbogXfjmSMDaKvG2ZQy0d4GziFWyLFWc0C4=";
    }.${stdenv.hostPlatform.system} or (throw "unsupported system ${stdenv.hostPlatform.system}");
```

Or disable the test on platforms not `x86_64-linux`. I tend to the do the later, as it would be hard to fix further issues with aarch64-linux we might encounter, and I assume that both of us are losing patience :).

---

Currently openjdk doesn't build for me in cross, so I don't think we'll be getting around to running the tests (or the application itself) in cross any time soon anyway. I don't have a big stake in this but I would recommend indeed just disabling the tests for cross.

---

> Currently openjdk doesn't build for me in cross, so I don't think we'll be getting around to running the tests (or the application itself) in cross any time soon anyway. I don't have a big stake in this but I would recommend indeed just disabling the tests for cross.

We are not talking about cross compilation from x86_64-linux to aarch64-linux, but rather native building of the NixOS test derivation on aarch64-linux. `openjdk` builds fine on aarch64-linux according to [`hydra-check`](https://github.com/nix-community/hydra-check).

---

Oh, wow. Why could the hash be different in the first place? I don't have an opinion here so I'd follow @doronbehar suggestions to only support `x86_64-linux`. I would disable the test within the `nextflow`-derivation itself (i.e. `passthru.tests.default = if host platform ...`), right?

---

> I would disable the test within the `nextflow`-derivation itself (i.e. `passthru.tests.default = if host platform ...`), right?

That would make ofborg avoid building that derivation when it looks up `nextflow.passthru.tests`, but still your commit `nixos/nextflow` will make at least Hydra try to build that test on aarch64-linux. You can use the `handleTestOn` function that is used in `all-tests.nix`.

---

I pushed a new commit using `handleTestOn`.

---

I think the hash is different because the data on the docker image differs among architectures/systems. Maybe one could have both hashes, depending on the `$system`. That might be something for the future.